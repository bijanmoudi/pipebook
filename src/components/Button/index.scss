/*=====================================
=            Button Styles            =
=====================================*/

@import '../../assets/styles/constants/layout';
@import '../../assets/styles/helpers/colors';
@import '../../assets/styles/helpers/units';

.button {
	display: inline-block;
	vertical-align: middle;
	min-width: em(95, 15);
	padding: rem(10) rem(20);
	border-radius: rem($border-radius);
	border: rem(1) solid get-color('gray-light', 0.5);
	box-shadow: 0 rem(1) rem(2) get-color('gray', 0.25);
	background-color: get-color('white');
	color: inherit;
	text-decoration: none;
	text-align: center;
	line-height: 1;
	font-size: em(15);
	font-weight: $semi-bold-font-weight;
	appearance: none;
	user-select: none;
	&:hover,
	&:focus {
		background-color: get-color('gray-light', 0.05);
	}
	&:hover {
		border-color: get-color('gray-light');
	}
	&:focus {
		box-shadow: 0 0 0 rem(1);
		border-color: get-color('gray-dark');
	}
	&--small,
	&--xsmall {
		min-width: auto;
		box-shadow: 0 rem(1) rem(3) get-color('gray', 0.5);
		padding: rem(4) rem(6);
		font-size: em(12);
	}
	&--xsmall {
		font-size: em(8);
	}
	&--green {
		border-color: get-color('green');
		color: get-color('green');
		&:hover,
		&:focus {
			border-color: get-color('green');
			background-color: get-color('green', 0.025);
		}
	}
	&--red {
		border-color: get-color('red');
		color: get-color('red');
		&:hover,
		&:focus {
			border-color: get-color('red');
			background-color: get-color('red', 0.025);
		}
	}
	&[disabled],
	&.is--disabled {
		opacity: 0.5;
		filter: grayscale(1);
		cursor: not-allowed;
	}
}
