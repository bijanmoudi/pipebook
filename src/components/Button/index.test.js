import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

const props = {
	size: 'small',
	color: 'green',
	text: 'Click Me',
	onClick: () => {},
	children: <button onClick={() => {}}>Click Me!</button>
};

describe('Testing "Button" component', () => {
	it('Rendering wrapping elements', () => {
		const tree = renderer
			.create(<Button size={props.size}>{props.children}</Button>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Rendering creating element', () => {
		const tree = renderer
			.create(
				<Button onClick={props.onClick} color={props.color}>
					Click Me!
				</Button>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
