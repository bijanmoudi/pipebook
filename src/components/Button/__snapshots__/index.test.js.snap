// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing "Button" component Rendering creating element 1`] = `
<button
  className="button button--green"
  onClick={[Function]}
>
  Click Me!
</button>
`;

exports[`Testing "Button" component Rendering wrapping elements 1`] = `
<button
  className="button button--small"
  onClick={[Function]}
>
  Click Me!
</button>
`;
